### PR TITLE
fix: return error if user not found but identity exists

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Currently, if an identity exists without a user (e.g. the dev accidentally inserts a user identity without creating a user), `IsDuplicatedEmail` function will panic because we omit the check for `IsNotFound(err)`. This leads to a `nil` user being accessed when you check for `user.Aud` which leads to a panic.

### CODE REVIEW

File: internal/models/user.go [hunk 1]
```diff
@@ -573,10 +573,9 @@ func IsDuplicatedEmail(tx *storage.Connection, email, aud string, currentUser *U
 
 	for _, userID := range userIDs {
 		user, err := FindUserByID(tx, userID)
-		if err != nil && !IsNotFoundError(err) {
+		if err != nil {
 			return nil, errors.Wrap(err, "unable to find user from email identity for duplicates")
 		}
-
 		if user.Aud == aud {
```
Comment (hf) [inline internal/models/user.go:579]: Why this change? Shouldn't it be more like:

```
if err != nil && !IsNotFoundError(err) {
// ...
}

if user != nil && user.Aud == aud {
// ...
```

Comment (kangmingtay) [reply internal/models/user.go:579]: that would swallow the error though - we want it to return an error if there's an identity but no user found

Comment (kangmingtay) [reply internal/models/user.go:579]: it shouldn't be possible for an identity to exist without a user

Comment (J0) [reply internal/models/user.go:579]: I think the current state is fine. If we want to consider the approach of `user != nil && user.Aud == aud` I guess another option could be:

```
if err != nil && !IsNotFoundError(err) {
// ...
// same as err != nil && IsNotFoundError(err)
// which should have the same effect as current commit except more verbose
} else if IsNotFoundError(err) [
// ...
}

if user != nil && user.Aud == aud {
// ...
```
